Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodeMarkerDetector.cs

```diff
@@ -0,0 +1,257 @@
+#endif
+        }
+
+        /// <inheritdoc />
+        public bool TryGetMarkerSize(int markerId, out float size)
+        {
+            bool found = false;
+            size = 0.0f;
+            lock(_cachedMarkers)
+            {
+                foreach (var marker in _cachedMarkers)
+                {
+                    if (marker.Value.MarkerId == markerId)
+                    {
+                        size = marker.Value.MarkerSize;
+                        found = true;
+                        break;
+                    }
+                }
+            }
+
+            return found;
+        }
+
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+        protected void Start()
+        {
+            if (_qrCodesManager == null)
+            {
+                _qrCodesManager = QRCodesManager.FindOrCreateQRCodesManager(gameObject);
+                _qrCodesManager.QRCodeAdded += QRCodeAdded;
+                _qrCodesManager.QRCodeRemoved += QRCodeRemoved;
+                _qrCodesManager.QRCodeUpdated += QRCodeUpdated;
+            }
+
+            var result = _qrCodesManager.StartQRTracking();
+            Debug.Log("Started qr tracker: " + result.ToString());
+        }
+
+        protected void Update()
+        {
+            if (_tracking &&
+                _processMarkers)
+            {
+                ProcessMarkerUpdates();
+            }
+        }
+
+        private void QRCodeAdded(object sender, QRCodeEventArgs<QRCodesTrackerPlugin.QRCode> e)
+        {
+            int markerId;
+            lock (_cachedMarkers)
+            {
+                if (!_cachedMarkers.ContainsKey(e.Data.Id) &&
+                    TryGetMarkerId(e.Data.Code, out markerId))
+                {
+                    _cachedMarkers.Add(e.Data.Id, new QRCodeInfo(e.Data.Id, markerId, e.Data.PhysicalSizeMeters));
+                    _processMarkers = true;
+                }
+            }
+        }
+
+        private void QRCodeUpdated(object sender, QRCodeEventArgs<QRCodesTrackerPlugin.QRCode> e)
+        {
+            int markerId;
+            lock (_cachedMarkers)
+            {
+                if (!_cachedMarkers.ContainsKey(e.Data.Id) &&
+                TryGetMarkerId(e.Data.Code, out markerId))
+                {
+                    _cachedMarkers.Add(e.Data.Id, new QRCodeInfo(e.Data.Id, markerId, e.Data.PhysicalSizeMeters));
+                    _processMarkers = true;
+                }
+            }
+        }
+
+        private void QRCodeRemoved(object sender, QRCodeEventArgs<QRCodesTrackerPlugin.QRCode> e)
+        {
+            lock(_cachedMarkers)
+            {
+                if (_cachedMarkers.ContainsKey(e.Data.Id))
+                {
+                    _cachedMarkers.Remove(e.Data.Id);
+                    _processMarkers = true;
+                    if (_markerCoordinateSystems.ContainsKey(e.Data.Id))
+                    {
+                        _markerCoordinateSystems.Remove(e.Data.Id);
```

Comment:
You don't need to guard this call with a ContainsKey, Remove should return true/false upon succesful deletion